runtime/interpreter/interpreter.go

```diff
@@ -3026,6 +3000,58 @@ var stringFunction = func() Value {
 		),
 	)
 
+	addMember(
+		sema.StringTypeFromUtf8FunctionName,
+		NewUnmeteredHostFunctionValue(
+			func(invocation Invocation) Value {
+				argument, ok := invocation.Arguments[0].(*ArrayValue)
+				if !ok {
+					panic(errors.NewUnreachableError())
+				}
+
+				inter := invocation.Interpreter
+
+				// meter byte count, not utf-8 codepoints
+				memoryUsage := common.NewArrayExpressionMemoryUsage(
+					argument.Count(),
+				)
+
+				iter, err := argument.array.Iterator()
+
+				if err != nil {
+					panic(errors.NewExternalError(err))
+				}
+
+				buf := make([]byte, argument.array.Count())
+
+				idx := 0
+				// naively read the entire byte stream into a byte array
+				for {
```

Comment:
I'd use the `iter.Iterate` function here instead of a while loop. If you don't want to, I'd still suggest moving the checks into the loop header like `for val, err := iter.Next(); val != nil; val, err = iter.Next()` instead of having them in the body.